substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -879,24 +1415,26 @@ impl<T: Config> Pallet<T> {
 
         // validation
         for report in &reports {
-            if !Contracts::<T>::contains_key(report.contract_id) {
-                continue;
-            }
-            if !ContractBillingInformationByID::<T>::contains_key(report.contract_id) {
+            if !Deployments::<T>::contains_key(report.contract_id) {
                 continue;
             }
-
             // we know contract exists, fetch it
             // if the node is trying to send garbage data we can throw an error here
-            let contract =
-                Contracts::<T>::get(report.contract_id).ok_or(Error::<T>::ContractNotExists)?;
-            let node_contract = Self::get_node_contract(&contract)?;
+            let deployment =
+                Deployments::<T>::get(report.contract_id).ok_or(Error::<T>::DeploymentNotExists)?;
+            let contract_cr = Contracts::<T>::get(deployment.capacity_reservation_id)
+                .ok_or(Error::<T>::CapacityReservationNotExists)?;
+            let capacity_reservation_contract =
+                Self::get_capacity_reservation_contract(&contract_cr)?;
             ensure!(
```

Comment:
Don't throw errors here, rather ignore contracts that don't exist anymore. Otherwise the entire batch will fail and ZOS can be out of sync and push reports for contracts that are already deleted..

Reply:
It should actually never throw the error. There is an if right before it.